src/fsipy/simulations/simulation_common.py

```diff
@@ -0,0 +1,274 @@
+              .format(combined_info['num_cells'] / volume))
+        print()
+
+
+def load_mesh_info(mesh_path: str) -> Tuple[List[int], List[int], int, float, List[float], List[float]]:
+    """
+    Load and process mesh information from a JSON file.
+
+    Args:
+        mesh_path (str): The path to the mesh file.
+
+    Returns:
+        Tuple[List[int], List[int], int, float, List[float], List[float]]:
+            A tuple containing:
+            - id_in (List[int]): List of inlet IDs.
+            - id_out (List[int]): List of outlet IDs.
+            - id_wall (int): Computed wall ID.
+            - Q_mean (float): Mean flow rate.
+            - area_ratio (List[float]): List of area ratios.
+            - area_inlet (List[float]): List of inlet areas.
+
+    Raises:
+        FileNotFoundError: If the info file is not found.
+    """
+    info_file = Path(mesh_path).with_name(Path(mesh_path).stem + "_info.json")
+
+    try:
+        # Read and parse the JSON info file
+        with open(info_file) as f:
+            info = json.load(f)
+    except FileNotFoundError:
+        raise FileNotFoundError(f"Info file '{info_file}' not found.")
+
+    # Extract information from the JSON data
+    id_in = info['inlet_id']
+    id_out = info['outlet_ids']
+    id_wall = min(id_in + id_out) - 1
+    Q_mean = info['mean_flow_rate']
+    area_ratio = info['area_ratio']
+    area_inlet = info['inlet_area']
+
+    return id_in, id_out, id_wall, Q_mean, area_ratio, area_inlet
+
+
+def load_probe_points(mesh_path: Union[str, Path]) -> np.ndarray:
+    """
+    Load probe points from a corresponding file based on the mesh file's path.
+
+    Args:
+        mesh_path (str or Path): The path to the mesh file.
+
+    Returns:
+        np.ndarray: An array containing the loaded probe points.
+    """
+    mesh_path = Path(mesh_path)
+    rel_path = mesh_path.stem + "_probe_point"
+    probe_points_path = mesh_path.parent / rel_path
+    probe_points = np.load(probe_points_path, encoding='latin1', fix_imports=True, allow_pickle=True)
+
+    return probe_points
+
+
+def print_probe_points(v: Function, p: Function, probe_points: List[Union[float, np.ndarray]]) -> None:
+    """
+    Print velocity and pressure at probe points.
+
+    Args:
+        v (dolfin.Function): Velocity function with components v.sub(0), v.sub(1), and v.sub(2).
+        p (dolfin.Function): Pressure function.
+        probe_points (list): List of probe points.
+
+    Returns:
+        None
+    """
+    for i, point in enumerate(probe_points):
+        # Extract components of velocity and pressure at the probe point
+        uu = peval(v.sub(0), point)
+        vv = peval(v.sub(1), point)
+        ww = peval(v.sub(2), point)
+        pp = peval(p, point)
+
+        if MPI.rank(MPI.comm_world) == 0:
+            print(f"Probe Point {i}: Velocity: ({uu}, {vv}, {ww}) | Pressure: {pp}")
+
+
+def peval(f: Function, x: Union[float, np.ndarray]) -> np.ndarray:
```

Comment:
Are we here assuming that the probe points `x` are the same on all processes? what happens if they are not? Then the allreduce doesn't work I guess?

Reply:
Yes, the probe point `x` are the same on all processes.